### PR TITLE
Space creation element-ios

### PR DESCRIPTION
**Pull request by gileluard**
Needed for https://github.com/vector-im/element-ios/issues/5224

### CODE REVIEW

File: MatrixSDK/Contrib/Swift/MXSession.swift [hunk 1]
```diff
@@ -137,8 +137,82 @@ public extension MXSession {
...
+            let guestAccessStateEvent = stateEventBuilder.buildGuestAccessEvent(withAccess: .forbidden)
+            parameters.addOrUpdateInitialStateEvent(guestAccessStateEvent)
+            
+            let historyVisibilityStateEvent = stateEventBuilder.buildHistoryVisibilityEvent(withVisibility: .shared)
+            parameters.addOrUpdateInitialStateEvent(historyVisibilityStateEvent)
+            
+            let joinRuleSupportType = homeServerCapabilities.isFeatureSupported(.restricted)
+            if joinRuleSupportType == .supported || joinRuleSupportType == .supportedUnstable {
+                guard let parentRoomId = parentRoomId else {
+                    fatalError("[MXSession] createRoom: parentRoomId is required for restricted room")
```
[the remark this answers is not in the document]
Comment (gileluard) [reply MatrixSDK/Contrib/Swift/MXSession.swift:186]: I would disagree in this case. a `nil`` parentRoomId` is typically a mistake from the developer not something that can happen with the environment. Don't forget that `parentRoomId` is not mandatory for `private` and `public` rooms.

File: MatrixSDKTests/MXSpaceChildContentTests.swift [hunk 2]
```diff
@@ -158,7 +192,24 @@ class MXSpaceChildContentTests: XCTestCase {
                                             bobSession.checkVisibilityOf(of: createdRoom, in: createdSpace!, expectedCount: 1) {
                                                 // Checking the visibility of the room from Alice's POV
                                                 aliceSession.checkVisibilityOf(of: createdRoom, in: spaceFromAlice, expectedCount: expectedCount) {
-                                                    expectation.fulfill()
+                                                    if testRemoveRoom {
+                                                        //Rremove the room from the space
```
Comment (manuroe) [inline MatrixSDKTests/MXSpaceChildContentTests.swift:196]:
```suggestion
                                                        //Remove the room from the space
```

File: MatrixSDK/Contrib/Swift/MXSession.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (manuroe) [reply MatrixSDK/Contrib/Swift/MXSession.swift:186]: Got it. This is an old good `NSParameterAssert`

**Review by manuroe — APPROVED** [review]
This is all good to me.
Accept my change on a typo and you can merge it.

**Pull request merged**